### PR TITLE
Cytoscape: Add fcose-layout types

### PR DESCRIPTION
**Pull request by Phpaulsen**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: (https://github.com/iVis-at-Bilkent/cytoscape.js-fcose)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## cytoscape ([<kbd>unpkg</kbd>](https://unpkg.com/browse/cytoscape@latest/))
was missing the following properties:
1. version
1. stylesheet
1. Stylesheet
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6bc59089ecdb092d77f022d94d0244270d38ca72
</p>

**Review by weswigham — CHANGES_REQUESTED** [review]
Since this layout option is added by an external package to `cytoscape`, the types should _probably_ also get added by a matching external package. Minimally, base cystoscape won't support this option from what I understand, so this is clearly wrong as-is.

File: types/cytoscape-fcose/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        quality?: "default" | "draft" | "proof";
+        // Use random node positions at beginning of layout
+        // if this is set to false, then quality option must be "proof"
+        randomize?: boolean;
+        // Whether or not to animate the layout
+        animate?: boolean;
+        // Duration of animation in ms, if enabled
+        animationDuration?: number;
+        // Easing of animation, if enabled
+        animationEasing?: undefined | "ease-out";
```
Comment (peterblazejewicz) [inline types/cytoscape-fcose/index.d.ts:56]:
```suggestion
        animationEasing?: "ease-out" | undefined;
```
just a convention here

File: types/cytoscape-fcose/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        piTol?: number;
+
+        /* incremental layout options */
+
+        // Node repulsion (non overlapping) multiplier
+        nodeRepulsion?: number;
+        // Ideal edge (non nested) length
+        idealEdgeLength(edge: any): number;
+        // Divisor to compute edge forces
+        edgeElasticity(edge: any): number;
```
Comment (peterblazejewicz) [inline types/cytoscape-fcose/index.d.ts:88]:
```suggestion
        idealEdgeLength?(edge: any): number;
        // Divisor to compute edge forces
        edgeElasticity?(edge: any): number;
```
both of those seem to be optional (from readme defaults):
you can rewrite them to `idealEdgeLength?: (edge: any) => number;`

Comment (Phpaulsen) [reply types/cytoscape-fcose/index.d.ts:88]: Actually had these changes in my own project. Frogot to copy them. 
Will be ready in a sec.

**Comment by Phpaulsen** [comment]
Are there any updates on merging this

**Comment by Phpaulsen** [comment]
Accidentally closed the pull request. Can someone reopen it?

**Pull request closed without merging**